File: models/appointment.go
package models

import "gorm.io/gorm"

type Appointment struct {
	gorm.Model
	UserID    uint   `json:"user_id"`
	CarID     uint   `json:"car_id"`
	Date      string `json:"date"`
	Comment   string `json:"comment"`
	ServiceID uint   `json:"service_id"`
	Time      string `json:"time"`

	User    User    `gorm:"foreignKey:UserID"`
	Car     Car     `gorm:"foreignKey:CarID"`
	Service Service `gorm:"foreignKey:ServiceID"`
}

File: static/appointment.js
const token = localStorage.getItem("token");

async function loadCars() {
    const res = await fetch("/api/cars", {
        headers: { 'Authorization': `Bearer ${token}` }
    });
    const cars = await res.json();
    const select = document.getElementById("carSelect");
    cars.forEach(car => {
        const opt = document.createElement("option");
        opt.value = car.ID;
        opt.textContent = `${car.brand} ${car.model}`;
        select.appendChild(opt);
    });
}

document.getElementById("appointmentForm").addEventListener("submit", async function (e) {
    e.preventDefault();
    const carId = document.getElementById("carSelect").value;
    const date = document.getElementById("date").value;
    const comment = document.getElementById("comment").value;

    const res = await fetch("/api/appointment", {
        method: "POST",
        headers: {
            "Authorization": `Bearer ${token}`,
            "Content-Type": "application/json"
        },
        body: JSON.stringify({ car_id: Number(carId), date, comment })
    });

    const data = await res.json();
    const msg = document.getElementById("message");
    if (res.ok) {
        msg.textContent = "Вы успешно записались!";
        msg.style.color = "green";
    } else {
        msg.textContent = data.error;
        msg.style.color = "red";
    }
});

loadCars();
